### DISCUSSION TITLE
Example for bulk loading using Python?

### DISCUSSION THREAD

**Discussion by alexpdp7** [chdb-io/chdb#287] [Q&A]
Is there any example about how to dump data into a Clickhouse-format database using Python? I have a generator of columns I'd like to load in bulk, and I wouldn't like to go over a dataframe or anything other...

edit: yeah, using chdb locally, not using a Clickhouse server.

**Comment by Gao123861** [comment]
To bulk load data from a CSV file into a ClickHouse format database using Python, you can use the clickhouse-driver library. Here is an example code that demonstrates how to bulk load data from a CSV file into ClickHouse:
 
from clickhouse_driver import Client
client = Client('host', user='username', password='password', database='database')
csv_file_path = 'example.csv'
table_name = 'your_table_name
client.execute(f'INSERT INTO {table_name} FORMAT CSVWithNames', csv_file_path)

**Reply by alexpdp7** [reply]
Thanks for the comment. However, as far as I know, `clickhouse_driver` cannot be used with chdb?

What I'm currently doing is a bit janky:

https://github.com/alexpdp7/news-rss/blob/62d90bc1fe9c08256d230783c05a3ae5510e1815/src/news_rss/__init__.py#L50

I create a CSV file and load it using `insert into ... select * from file(...)`.
